### PR TITLE
[PF-2422] Add getResource endpoint

### PR DESCRIPTION
**Pull request by rogerwangcs**
Expose the getResource method in ResourceDao as a new resource endpoint

### CODE REVIEW

File: openapi/src/parts/resource.yaml [hunk 1]
```diff
@@ -30,6 +30,31 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
+  /api/workspaces/v1/{workspaceId}/resources/{resourceId}:
```
Comment (yuhuyoyo) [inline openapi/src/parts/resource.yaml:33]: why is the existing get method not good enough for your usage?

Comment (rogerwangcs) [reply openapi/src/parts/resource.yaml:33]: The [axon server getFile endpoint](https://docs.google.com/document/d/1_6ruIYYzTnbfrqfjEquRsWUx1ZsNJSzCALa9iMZ6au0/edit?resourcekey=0-L8V4uz_kouNZoUhvjO1R_A#heading=h.mo0z046y10jz) accepts general resource id. We need this endpoint to determine the resourceType and validate whether it is a file containing resource. Without this endpoint, we would unecessarily get all resources in the workspace with enumerateResources then filter for the desired resource.

**Review by ddietterich — APPROVED** [review]
LGTM

File: openapi/src/parts/resource.yaml (hunk 1 above)
Comment (ddietterich) [reply openapi/src/parts/resource.yaml:33]: @yuhuyoyo I don't think there is an existing method that does this. Is there and I missed it?

Comment (yuhuyoyo) [reply openapi/src/parts/resource.yaml:33]: no. I meant we have endpoint per resources. i remember it was an intentional decision to not use the union attributes and implement endpoint per resources.

With this endpoint, why do we need GET per resource type at all?

i get why this is useful for the axon use case.

Comment (ddietterich) [reply openapi/src/parts/resource.yaml:33]: Good point. When we do the "v2" interface design, maybe we skip the per-resource GET in favor of this style!?

**Review by yuhuyoyo — APPROVED** [review]

**Pull request merged**